CMSgov/easi-app
# [EASI-2754] - Table state persistence

# EASI-2754

## Changes and Description

- Added context for storing/removing table state
  - Context state is refreshed on absence of IT Gov routes `/governance-review-team`
- Added method to persist page size in local storage for more permanent persistence
- Added hook `useTableState` for setting and storing state for react-table
- Updated test

Persisted Context State:
Active table (open/closed)
Page index
Column Sort
Query/global filter

Persisted Local Storage State:
Page size

As is, local storage page size is set to a single string key, may want to develop this further if we find users want different page sizes for difference tables.  

This state management is only configured for IT Gov admin home table, but can be expanded in the future if needed

## How to test this change

- Log in as IT Gov admin
- Change page index, sort, query, or active table
- Navigate to an admin view of request
- Navigate to different pages within the request
- Return to home table and verify state was persisted and set
- Navigate to any other page besides IT Gov request and return to. home table
- Verify state has been reset
Page size is the only state to persist in local storage on selection change
- Change page size selection and immediate refresh the browser to verify state persisted and set

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
